main.go

```diff
@@ -136,6 +140,12 @@ func cmd() *cobra.Command {
 		},
 		Run: func(_ *cobra.Command, args []string) {
 			err := upgrade.UpgradeProvider(context, repoOrg, repoName)
+			if err != nil {
+				msg, err := createFailureIssue(context, repoOrg, repoName)
+				if err != nil && context.InferVersion {
+					fmt.Println(msg)
+				}
```

Comment:
This doesn't make any sense to me. If we try to create the issue and fail, we should always report to the user. 

If we're gating issue creation on a Pulumi provider, the `&& context.InferVersion` check should be on line 143, before we try to create an issue.

Reply:
You are correct